### PR TITLE
WIP: Deprecate SbStringFormatWide function

### PR DESCRIPTION
**Pull request by maxz-lab**
Replace SbStringFormatWide() calls with POSIX vsnprintf()

b/307941391

Change-Id: Ic630572a66e6f637d1fbcc0eb7f0cb303d96bc12

### CODE REVIEW

File: starboard/common/string.h [hunk 1]
```diff
@@ -20,6 +20,9 @@
 #define STARBOARD_COMMON_STRING_H_
 
 #include <stdarg.h>
+#if SB_API_VERSION == 16
```
Comment (andrewsavage1) [inline starboard/common/string.h:23]: Shouldn't this be a `>=` check?

File: base/strings/string_util_starboard.h [hunk 2]
```diff
@@ -54,11 +54,13 @@ inline int strncmp16(const char16* s1, const char16* s2, size_t count) {
 #endif
 }
 
+#if SB_API_VERSION <= 15
 inline int vswprintf(wchar_t* buffer, size_t size,
                      const wchar_t* format, va_list arguments) {
   DCHECK(base::IsWprintfFormatPortable(format));
   return SbStringFormatWide(buffer, size, format, arguments);
```
Comment (y4vor) [inline base/strings/string_util_starboard.h:61]: We should remove the use of the API everywhere and switch to the POSIX equivalent.
In musl we should take care of backwards compatibility for older SB versions for Evergreen/Modular platforms.
And for win32 based platforms we'll have to add the POSIX API if it is missing.

File: starboard/shared/stub/string_format_wide.cc [hunk 3]
```diff
@@ -14,9 +14,11 @@
 
 #include "starboard/common/string.h"
 
+#if 0
```
Comment (andrewsavage1) [inline starboard/shared/stub/string_format_wide.cc:17]: I assume this should be a SB_API_VERSION check (if not, fully remove this block instead)

File: starboard/string.h [hunk 4]
```diff
@@ -20,6 +20,9 @@
 #define STARBOARD_STRING_H_
 
 #include <stdarg.h>
+#if SB_API_VERSION == 16
```
Comment (andrewsavage1) [inline starboard/string.h:23]: `>=`

File: starboard/string.h [hunk 5]
```diff
@@ -126,6 +129,7 @@ static SB_C_INLINE int SbStringFormatUnsafeF(char* out_buffer,
 // |buffer_size|: The size of |out_buffer|.
 // |format|: A string that specifies how the data should be formatted.
 // |arguments|: Variable arguments used in the string.
+#if SB_API_VERSION <= 15
```
Comment (y4vor) [inline starboard/string.h:132]: The typical convention is to use SB_API_VERSION < 16 when deprecating in SB 16.

File: base/strings/string_util_starboard.h [hunk 6]
```diff
@@ -37,6 +37,7 @@ inline int strncasecmp(const char* string1, const char* string2, size_t count) {
   return SbStringCompareNoCaseN(string1, string2, count);
 }
 
+#if SB_API_VERSION < 16
```
Comment (y4vor) [inline base/strings/string_util_starboard.h:40]: Let's remove this section.

File: base/strings/string_util_starboard.h [hunk 7]
```diff
@@ -54,11 +56,13 @@ inline int strncmp16(const char16* s1, const char16* s2, size_t count) {
 #endif
 }
 
+#if SB_API_VERSION < 16
```
Comment (y4vor) [inline base/strings/string_util_starboard.h:59]: Remove

File: starboard/client_porting/poem/stdio_poem.h [hunk 8]
```diff
@@ -28,7 +28,9 @@
 // and, out of compatibility concerns, we chose to not use
 // __VA_ARGS__ functionality.
 #undef vsnprintf
+#if SB_API_VERSION < 16
```
Comment (y4vor) [inline starboard/client_porting/poem/stdio_poem.h:31]: Remove that whole section

File: starboard/client_porting/poem/wchar_poem.h [hunk 9]
```diff
@@ -24,7 +24,9 @@
 #include "starboard/string.h"
 
 #undef vswprintf
+#if SB_API_VERSION < 16
```
Comment (y4vor) [inline starboard/client_porting/poem/wchar_poem.h:27]: remove

File: starboard/common/string.h [hunk 10]
```diff
@@ -20,6 +20,9 @@
 #define STARBOARD_COMMON_STRING_H_
 
 #include <stdarg.h>
+#if SB_API_VERSION >= 16
```
Comment (y4vor) [inline starboard/common/string.h:23]: remove all SbStringFormat instances

File: starboard/shared/starboard/log_raw_format.cc [hunk 11]
```diff
@@ -15,11 +15,18 @@
...
 #include <stdarg.h>
+#if SB_API_VERSION >= 16
+#include <stdio.h>
+#endif
 
 #include "starboard/common/string.h"
 
 void SbLogRawFormat(const char* format, va_list arguments) {
   char message[128];
+#if SB_API_VERSION < 16
```
Comment (y4vor) [inline starboard/shared/starboard/log_raw_format.cc:26]: Remove the 16 check SbStringFormat

File: third_party/boringssl/src/crypto/mem_starboard.c [hunk 12]
```diff
@@ -222,5 +222,9 @@ int BIO_snprintf(char *buf, size_t n, const char *format, ...) {
 }
 
 int BIO_vsnprintf(char *buf, size_t n, const char *format, va_list args) {
+#if SB_API_VERSION < 16
```
Comment (y4vor) [inline third_party/boringssl/src/crypto/mem_starboard.c:225]: remove the check and SbStringFormat

File: third_party/google_benchmark/src/colorprint_starboard.cc [hunk 13]
```diff
@@ -24,7 +27,11 @@ std::string FormatString(const char* msg, va_list args) {
   va_list args_copy;
   va_copy(args_copy, args);
 
+#if SB_API_VERSION < 16
```
Comment (y4vor) [inline third_party/google_benchmark/src/colorprint_starboard.cc:30]: remove the check and SbStringFormat

File: third_party/googletest/src/googletest/include/gtest/internal/gtest-port.h [hunk 14]
```diff
@@ -2102,7 +2106,11 @@ inline void Abort() { SbSystemBreakIntoDebugger(); }
 
 inline int VSNPrintF(char* out_buffer, size_t size, const char* format,
                       va_list args) {
+#if SB_API_VERSION < 16
```
Comment (y4vor) [inline third_party/googletest/src/googletest/include/gtest/internal/gtest-port.h:2109]: same

File: third_party/v8/src/base/platform/platform-starboard.cc [hunk 15]
```diff
@@ -345,7 +349,11 @@ int OS::SNPrintF(char* str, int length, const char* format, ...) {
 }
 
 int OS::VSNPrintF(char* str, int length, const char* format, va_list args) {
+#if SB_API_VERSION < 16
```
Comment (y4vor) [inline third_party/v8/src/base/platform/platform-starboard.cc:352]: Remove the guard and just use vsnprintf

**Pull request closed without merging**